### DISCUSSION TITLE
My goal is to make sequential difference gif file from this images.

### DISCUSSION THREAD

**Discussion by bemoregt** [ImageMagick/ImageMagick#2366] [Help]
HI, @alex @mirakui @yoya @rudchenkos @bgK 

I have converted an animated-gif file to sequence images.
My goal is to make sequential difference gif file from this images.

In other words, 
I have img1.png, img2.png, img3.png, ...imgN.png.
and 1st frame of result.gif is img2 - img1,
and 2nd frame of result.gif is img3 - img2, ...

How can I get this using IM ?

Thanks.
Best,
@bemoregt.

**Comment by dlemstra** [comment]
Next time please don't tag all those people @bemoregt, this is not the way to get an answer to your question.

**Reply by bemoregt** [reply]
Yes, I got it.

**Comment by rudchenkos** [comment]
@bemoregt I am not aware of any mechanism in ImageMagick to solve this. I would simply write a script in Perl/Python/Shell/whatever else which invokes ImageMagick's compare on the respective pairs

**Comment by snibgo** [comment]
The IM command to what you want could be:
```
magick ^
  -compose MinusDst ^
  ( img1.png img2.png -composite ) ^
  ( img2.png img3.png -composite ) ^
  ( img3.png img4.png -composite ) ^
  ...
  out.gif
```
(This is Windows syntax. Adjust for bash or whatever.)

You could create a script file that contains that command. Or create an IM script file that would then be called with "-script filename". Or an environment variable that contains the command, or part of it.
